server/handlers/events_streamer.go

```diff
@@ -475,3 +475,89 @@ func closeAdapterConnections(localMeshAdaptersLock *sync.Mutex, localMeshAdapter
 
 	return map[string]*meshes.MeshClient{}
 }
+
+// swagger:route POST /api/events EventsAPI idClientEventHandler
+// Receives client-generated events bound for the Notification Center.
+// responses:
+// 200:
+
+func (h *Handler) ClientEventHandler(w http.ResponseWriter, req *http.Request, prefObj *models.Preference, user *models.User, provider models.Provider) {
+	userID := uuid.FromStringOrNil(user.ID)
+
+	defer func() {
+		_ = req.Body.Close()
+	}()
+
+	var eventDetails map[string]interface{}
```

Comment:
this was to perform empty and type checks on the fields of the event before creating an event.

Reply:
Unmarshal it to events struct, so that you don't need to perform type casting for all values. Also if it is invalid/malformed event then it will get rejected at first validation itself.